File: app/src/main/java/com/example/pcmall/service/BrandService.java
package com.example.pcmall.service;

import com.example.pcmall.model.Brand;
import com.example.pcmall.model.response.ResponseResult;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

public interface BrandService {
    @FormUrlEncoded
    @POST("searchBrandByCid")
    Observable<ResponseResult<List<Brand>>> searchBrandByCid(@Field("cid") Integer cid);
}

File: core/network/src/main/java/com/example/pcmallcompose/core/network/service/BrandService.kt
package com.example.pcmallcompose.core.network.service

import com.example.pcmallcompose.core.model.Brand
import com.example.pcmallcompose.core.model.response.ResponseResult
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.POST

interface BrandService {
    @FormUrlEncoded
    @POST("searchBrandByCid")
    suspend fun searchBrandByCid(@Field("cid") cid: Int): ResponseResult<List<Brand>>
}